internal/monitors/collectd/hadoopjmx/hadoopjmx.go

```diff
@@ -0,0 +1,143 @@
+//
+// **hadoop-env.sh:**
+// ```
+// export HADOOP_NAMENODE_OPTS="-Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=5677 $HADOOP_NAMENODE_OPTS"
+// export HADOOP_DATANODE_OPTS="-Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=5679 $HADOOP_DATANODE_OPTS"
+// ```
+//
+// **yarn-env.sh:**
+// ```
+// export YARN_NODEMANAGER_OPTS="-Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=5678 $YARN_NODEMANAGER_OPTS"
+// export YARN_NODEMANAGER_OPTS="-Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=5678 $YARN_NODEMANAGER_OPTS"
+// ```
+//
+// This monitor has a set of built in MBeans configured for:
+// - [Name Nodes](https://github.com/signalfx/signalfx-agent/tree/master/internal/monitors/collectd/hadoopjmx/nameNodeMBeans.go)
+// - [Resource Manager](https://github.com/signalfx/signalfx-agent/tree/master/internal/monitors/collectd/hadoopjmx/resourceManagerMBeans.go)
+// - [Node Manager](https://github.com/signalfx/signalfx-agent/tree/master/internal/monitors/collectd/hadoopjmx/nodeManagerMBeans.go)
+// - [Data Nodes](https://github.com/signalfx/signalfx-agent/tree/master/internal/monitors/collectd/hadoopjmx/dataNodeMBeans.go)
+//
+// Sample YAML configuration:
+//
+//	Name Node
+// ```yaml
+// monitors:
+// - type: collectd/hadoopjmx
+//   host: 127.0.0.1
+//   port: 5677
+//   nodeType: nameNode
+// ```
+//
+// 	Resource Manager
+// ```yaml
+// monitors:
+// - type: collectd/hadoopjmx
+//   host: 127.0.0.1
+//   port: 5680
+//   nodeType: resourceManager
+// ```
+//
+// 	Node Manager
+// ```yaml
+// monitors:
+// - type: collectd/hadoopjmx
+//   host: 127.0.0.1
+//   port: 8002
+//   nodeType: nodeManager
+// ```
+//
+// 	Data Node
+// ```yaml
+// monitors:
+// - type: collectd/hadoopjmx
+//   host: 127.0.0.1
+//   port: 5679
+//   nodeType: dataNode
+// ```
+//
+// You may also configure the [collectd/hadoop](https://github.com/signalfx/signalfx-agent/tree/master/docs/monitors/collectd/hadoop)
+// monitor to collect additional metrics about the hadoop cluster from the REST API
+//
+
+// Config is the monitor-specific config with the generic config embedded
+type Config struct {
+	genericjmx.Config `yaml:",inline"`
+	// Hadoop Node Type
+	NodeType nodeType `yaml:"nodeType" validate:"required"`
+}
+
+// Monitor is the main type that represents the monitor
+type Monitor struct {
+	*genericjmx.JMXMonitorCore
+}
+
+// Configure configures the hadoopjmx monitor and instantiates the generic jmx
+// monitor
+func (m *Monitor) Configure(conf *Config) error {
+	// create the mbean map with the appropriate mbeans for the given node type
+	var newMBeans genericjmx.MBeanMap
+	switch conf.NodeType {
```

Comment:
The YAML decoder doesn't actually raise any errors if `nodeType` is not in your set of constants, so you might want to add a `Validate` method to the `Config` struct so that if a user makes a typo they don't get very confused on why nothing is working.